elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java

```diff
@@ -368,6 +368,27 @@ public void testSslServiceAuthTLS13() throws Throwable {
                 "OU=Elytron,O=Elytron,C=UK,ST=Elytron,CN=Firefly", "TLS_AES_256_GCM_SHA384", true);
     }
 
+    @Test
+    public void testSslServiceAuthSSLv2Hello() throws Throwable {
+        Assume.assumeFalse("Skipping testSslServiceAuthSSLv2Hello as IBM JDK does not support enabling SSLv2Hello " +
+                "in the client", JdkUtils.isIbmJdk());
+        testCommunication("ServerSslContextSSLv2Hello", "ClientSslContextSSLv2Hello", false, "OU=Elytron,O=Elytron,C=CZ,ST=Elytron,CN=localhost",
+                "OU=Elytron,O=Elytron,C=UK,ST=Elytron,CN=Firefly");
+    }
+
+    @Test
+    public void testSslServiceAuthProtocolMismatchSSLv2Hello() throws Throwable {
```

Comment:
It would also be good to add a test for the case where SSLv2Hello is only configured on the server side.

Reply:
Added

Reply:
Looks like it would be good to add one more test case that makes use of the default protocols on the server side and the client side to ensure that SSLv2Hello isn't enabled by default.

Reply:
It would also be good to either tweak `testCommunication` or create a new method that could be used instead in order to assert that `serverSocket.getEnabledProtocols()`, `clientSocket.getEnabledProtocols()`, `serverSocket.getSession().getProtocol()`, and `clientSocket.getSession().getProtocol()` return the expected values.

Reply:
I've modified testCommunication to check the configured protocols.